src/main/java/org/jabref/gui/ReplaceStringView.java

```diff
@@ -0,0 +1,184 @@
+
+    @FXML
+    public void initialize() {
+        visualizer.setDecoration(new IconValidationDecorator());
+        LimitFieldInput.setEditable(true);
+        FindField.setEditable(true);
+        ReplaceField.setEditable(true);
+        checkLimit.setSelected(false);
+    }
+
+    /**
+     * FXML Message handler
+    */
+    @FXML
+    public void buttonReplace() {
+        findString = FindField.getText();
+        replaceString = ReplaceField.getText();
+        fieldStrings = LimitFieldInput.getText().toLowerCase().split(";");
+        if ("".equals(findString))
+        {
+            exitSignal = true;
+            st.close();
+            return;
+        }
+        replace();
+        exitSignal = true;
+        st.close();
+    }
+
+    @FXML
+    public void buttonCancel() {
+        exitSignal = true;
+        st.close();
+    }
+
+    @FXML
+    public void radioAll() {
+        AllFieldReplace = true;
+    }
+
+    @FXML
+    public void radioLimit() {
+        AllFieldReplace = false;
+    }
+
+    @FXML
+    public void selectOnly() {
+        selOnly = !selOnly;
+    }
+
+    public boolean isExit() {
+        return this.exitSignal;
+    }
+
+    /**
+     *  General replacement, same as Action:Replace_All in BasePanel.java
+     *  Rep check: BasePanel == null
+     *  @return : replace count
+     */
+    public int replace() {
+        final NamedCompound ce = new NamedCompound(Localization.lang("Replace string"));
+        int counter = 0;
+        if (this.panel == null)
+            return 0;
+        if (this.selOnly) {
+            for (BibEntry bibEntry: this.panel.getSelectedEntries())
+            {
+                counter += replaceItem(bibEntry, ce);
+            }
+        }
+        else {
+            for (BibEntry bibEntry: this.database.getEntries())
+            {
+                counter += replaceItem(bibEntry, ce);
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * Does the actual operation on a Bibtex entry based on the
+     * settings specified in this same dialog. Returns the number of
+     * occurences replaced.
+     * Copied and Adapted from org.jabref.gui.ReplaceStringDialog.java
+     */
+    public int replaceItem(BibEntry be, NamedCompound ce) {
+        int counter = 0;
+        if (this.AllFieldReplace) {
+            for (String s : be.getFieldNames()) {
+                counter += replaceField(be, s, ce);
+            }
+        } else {
+            for (String fld : fieldStrings) {
+                counter += replaceField(be, fld, ce);
+            }
+        }
+        return counter;
+    }
+
+    private int replaceField(BibEntry be, String fieldname, NamedCompound ce) {
+        if (!be.hasField(fieldname)) {
+            return 0;
+        }
+        String txt = be.getField(fieldname).get();
+        StringBuilder sb = new StringBuilder();
+        int ind;
+        int piv = 0;
+        int counter = 0;
+        int len1 = this.findString.length();
```

Comment:
maybe better findStringLength to directly make clear what the variable represent